app/_kong_plugins/request-termination/examples/echo-response-to-client.yaml

```
@@ -0,0 +1,26 @@
+description: |
+  Block requests and echo the response back to the client. This is especially useful for debugging.
+
+  By combining the `config.echo` parameter with a header name specified in `config.trigger`, 
+  you can make the plugin trigger only on specific requests and avoid disturbing real traffic.
+
+title: 'Echo response back to client'
```

```suggestion
description: |
  Block requests and echo the response back to the {{site.base_gateway}}. This is especially useful for debugging.

  By combining the `config.echo` parameter with a header name specified in `config.trigger`, 
  you can make the plugin trigger only on specific requests and avoid disturbing real traffic.

title: 'Echo response back to {{site.base_gateway}}'
```
I don't know if this is correct, but "client" (sender of the request) feels wrong here. If this just echoes the 401 error back to the sender of the request, that feels like the same behavior as [Trigger on header](https://deploy-preview-527--kongdeveloper.netlify.app/plugins/request-termination/examples/trigger-on-header/).

Since this is used for debugging, it would make sense for it to echo the response to Kong so it would be picked up in logs.

---

> If this just echoes the 401 error back to the sender of the request, that feels like the same behavior as Trigger on header.

`trigger_on_header` triggers request termination if a specific header is present in a request. If the header isn't there, the request won't be terminated.

`echo` is a debugging tool - that's the echoing back to client. If it isn't enabled, then the _sender_ (eg you or me, or the originating application) of the request gets debugging info back.

Here's the response I get without `echo`:

```
{"message":"Error - Authentication required"}
```

And here's with `echo` enabled:
```
{"matched_service":{"updated_at":1740594382,"retries":5,"name":"example-service","protocol":"http","id":"a86f2a98-7c46-4398-9e3f-c09036cd1546","ws_id":"c36192b9-5019-483c-9d95-67f76bc70ed0","connect_timeout":60000,"read_timeout":60000,"enabled":true,"host":"blue","created_at":1740594378,"port":80,"write_timeout":60000},"matched_route":{"strip_path":true,"https_redirect_status_code":426,"updated_at":1740594378,"preserve_host":false,"protocols":["http","https"],"service":{"id":"a86f2a98-7c46-4398-9e3f-c09036cd1546"},"regex_priority":0,"id":"0d3527e1-167f-4ac3-8371-cd149730c1de","ws_id":"c36192b9-5019-483c-9d95-67f76bc70ed0","path_handling":"v0","paths":["/anything"],"created_at":1740594378,"response_buffering":true,"name":"example-route","request_buffering":true},"kong":{"node_id":"58950592-0ada-45d0-8e99-b62bcd41d55f","hostname":"926a1d013fa5","worker_pid":2557},"request":{"uri_captures":{"named":{},"unnamed":[]},"raw_body":"","query":{},"headers":{"user-agent":"curl/8.4.0","accept":"*/*","host":"localhost:8000"},"host":"localhost","port":8000,"path":"/anything","scheme":"http","method":"GET"},"message":"Error - Authentication required"}%   
```

---

Thanks for the info, your edits make sense to me!